### DISCUSSION TITLE
Approve your subscription appears on Android

### DISCUSSION THREAD

**Discussion by normenster** [hyochan/react-native-iap#2240] [Q&A]
thanks for the great library! We have a good implementation currently live. Nevertheless 2 clients mentioned that, after successfully subscribing, google play store requires them to confirm the subscription, and that this should be done in our app. Otherwise they will get refunded. I really don`t know what we have to do and how we handle this case. Anyones knows what to do with this dialog? 

(screenshot from android subscription manage page, the red text says "Confirm your subscription")

<img width="645" alt="Bildschirm­foto 2023-01-31 um 10 17 11" src="https://user-images.githubusercontent.com/1347907/217293499-62d0db9b-fed4-4335-808a-1285dcfe80d0.png">

**Comment by andresesfm** [comment]
Can you confirm that `finishTransaction` was called in those cases?

**Comment by normenster** [comment]
The Problem was that Users switched to background in the middle of the purchasing process. So finishedTransaction wasn`t called. I thought that the "checkCurrentPurchase" hook was meant to deal with it when the app came back to the foreground, but it somehow did not work. I ended up checking for pending purchases when the app comes back to the foreground and manually triggering finishTransaction afterwards if needed.
